### PR TITLE
Give button in main menu

### PR DESCRIPTION
**Pull request by RoyEJohnson**

### CODE REVIEW

File: src/app/components/shell/header/upper-menu/upper-menu.js [hunk 1]
```diff
@@ -1,32 +1,79 @@
...
+//                     this.model.showBlog = true;
+//                     this.update();
+//                 }
+//             } catch (e) {
+//                 console.log(e);
+//             }
+//         })();
+//     }
+//
+// }
```
Comment (nathanstitt) [inline src/app/components/shell/header/upper-menu/upper-menu.js:79]: commented out code could be removed.

File: src/app/helpers/controller/cms-mixin.js [hunk 2]
```diff
@@ -86,23 +85,12 @@ export async function fetchFromCMS(slug, preserveWrapping=false) {
...
-
 export async function fetchPageData({slug, preserveWrapping, setsPageTitleAndDescription=true}) {
     const data = await fetchFromCMS(slug, preserveWrapping);
 
     await loadImages(data);
     if (setsPageTitleAndDescription) {
-        setTitleAndDescription(data);
+        $.setPageTitleAndDescriptionFromBookData(data);
     }
     return data;
```
Comment (nathanstitt) [inline src/app/helpers/controller/cms-mixin.js:95]: same comment as above, might need to wrap this in useEffect

**Review by nathanstitt — APPROVED** [review]

File: src/app/components/jsx-helpers/jsx-helpers.jsx [hunk 3]
```diff
@@ -64,6 +62,22 @@ export function useCanonicalLink(controlsHeader=true) {
...
 
+export function LoaderPage({slug, Child, props={}, preserveWrapping, doDocumentSetup=false}) {
+    const [data, statusPage] = usePageData({slug, setsPageTitleAndDescription: false, preserveWrapping});
+
+    useCanonicalLink(doDocumentSetup);
+    if (statusPage) {
+        return statusPage;
+    }
+    if (doDocumentSetup) {
+        $.setPageTitleAndDescriptionFromBookData(data);
```
[the remark this answers is not in the document]
Comment (RoyEJohnson) [reply src/app/components/jsx-helpers/jsx-helpers.jsx:73]: It affects `<head>` elements. Not sure whether that counts. But `useEffect` is probably right.

File: src/app/components/shell/header/main-menu/main-menu.js [hunk 4]
```diff
@@ -1,60 +1,98 @@
...
-            items: [
-                {url: '/about', label: 'About Us'},
-                {url: '/team', label: 'Team'},
-                {url: '/research', label: 'Research'},
-                {url: '/institutional-partnership', label: 'Institutional Partnerships'},
-                {url: '/creator-fest', label: 'Creator Fest'}
-            ]
-        }));
-        attachLoginMenu(this.regions.loginMenu.el);
+    const dd = new Dropdown({
```
[the remark this answers is not in the document]
Comment (RoyEJohnson) [reply src/app/components/shell/header/main-menu/main-menu.js:17]: The promise could throw, but it exists outside `useDataFromPromise`. Inside, it should never throw.

File: src/app/helpers/controller/cms-mixin.js (hunk 2 above)
Comment (RoyEJohnson) [reply src/app/helpers/controller/cms-mixin.js:95]: This is not a hook. It is a helper that is used by a hook (and is called from inside `useEffect`)

File: src/app/components/jsx-helpers/jsx-helpers.jsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (nathanstitt) [reply src/app/components/jsx-helpers/jsx-helpers.jsx:73]: yeah it'd probably be ok, but it'd be more efficient with `useEffect` anyway.  What you *really* don't want to cause is a rendering loop where render kicks off something that will eventually trigger another render.

**Pull request merged**